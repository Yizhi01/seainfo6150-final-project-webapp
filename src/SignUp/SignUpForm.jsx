import React, { useState} from "react";
import styles from "./SignUpForm.module.css";

const SignUpForm = () => {
  const [submittedForm, setSubmittedForm] = useState();

  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  }

  return (
    <div>
      {submittedForm ? (
        <div className={styles.successPage} >
          <img className={styles.successImage}
            src="http://source.unsplash.com/qWlkCwBnwOE/1000x619" alt="funny pineapples" />
          Welcome {submittedForm.get("username")}, you have signed up successfully. Enjoy exploring our website!
        </div>
      ) : (
        <form className={styles.signUpForm} onSubmit={onSubmit}>
          <h1>Tasty Food</h1>
          <h3>Create an Account</h3>
          <div className={styles.formGroup}>
            <label htmlFor="email">Email</label>
            <input className={styles.input} type="email" name="email" id="email" required />
          </div>
          <div className={styles.formGroup}>
            <label htmlFor="username">Username ((only letters and numbers)</label>
            <input className={styles.input} type="text" name="username" id="username" 
              pattern="[A-Za-z0-9]{4,20}" title="Please enter 4-20 letters or numbers" required />
          </div>
          <div className={styles.formGroup}>
            <label htmlFor="password">Password</label>
            <input className={styles.input} type="password" name="password" id="password" required />
          </div>
          <div className={styles.formGroup}>
            <input className={styles.signUpBtn} type="submit" value="Sign Up"/>
          </div>
          
        </form>
      )}
    </div>
  )
}

export default SignUpForm;