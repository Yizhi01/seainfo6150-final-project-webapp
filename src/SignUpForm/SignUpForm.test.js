import React from "react";
import { render } from "@testing-library/react";
import SignUpForm from "./SignUpForm.jsx";

it("renders the SignUpForm component correctly", () => {
  const { container } = render(
    <SignUpForm />
  );
  expect(container).toMatchSnapshot();
});
