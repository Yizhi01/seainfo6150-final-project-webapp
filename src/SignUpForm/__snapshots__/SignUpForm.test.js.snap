// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SignUpForm component correctly 1`] = `
<div>
  <div>
    <form
      class="signUpForm"
    >
      <h1>
        Tasty Food
      </h1>
      <h3>
        Create an Account
      </h3>
      <div
        class="formGroup"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          class="input"
          id="email"
          name="email"
          placeholder="example@gmail.com"
          required=""
          type="email"
        />
      </div>
      <div
        class="formGroup"
      >
        <label
          for="username"
        >
          Username ((only letters and numbers)
        </label>
        <input
          class="input"
          id="username"
          name="username"
          pattern="[A-Za-z0-9]{4,20}"
          required=""
          title="Please enter 4-20 letters or numbers"
          type="text"
        />
      </div>
      <div
        class="formGroup"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          class="input"
          id="password"
          name="password"
          required=""
          type="password"
        />
      </div>
      <div
        class="formGroup"
      >
        <input
          class="signUpBtn"
          type="submit"
          value="Sign Up"
        />
      </div>
    </form>
  </div>
</div>
`;
